osquery/tables/system/windows/processes.cpp

```diff
@@ -299,13 +361,23 @@ QueryData genProcesses(QueryContext& context) {
       query += " WHERE " + boost::algorithm::join(constraints, " OR ");
     }
   }
+  std::map<std::int32_t, std::map<std::string, std::int64_t>> perfData;
```

Comment:
Rather than making a `map`, populating it, and then sending it along in a function call to `genProcess`, could you change how `genProcess` is working to instead take in a `Row&`? Then you could generate this data, check if it's used in the `genProcess` loop below, and just populate the `Row` in the for loop, as opposed to passing this data structure to the `genProcess` function.

Reply:
When generating the `Row&`  i would need to include the pid in the row so that i can later add more data to it. I would need to put in few more `isAnyColumnUsed` conditionals elsewhere to account for if we're not selecting any perf data and populate pid and such in `genProcess`. Sounds like it could make things a bit more complicated, do you think?